src/envoy/http/authn/authn_utils.cc

```diff
@@ -22,47 +22,31 @@ namespace Http {
 namespace Istio {
 namespace AuthN {
 namespace {
-class JsonIterCallback : public Logger::Loggable<Logger::Id::filter> {
- public:
-  // The json iterator callback function
-  static bool JsonIteratorCallback(const std::string&,
-                                   const Envoy::Json::Object&,
-                                   istio::authn::JwtPayload* payload);
-};
-
-bool JsonIterCallback::JsonIteratorCallback(const std::string& key,
-                                            const Envoy::Json::Object& obj,
-                                            istio::authn::JwtPayload* payload) {
-  ENVOY_LOG(debug, "{}: key is {}", __FUNCTION__, key);
-  ::google::protobuf::Map< ::std::string, ::std::string>* claims =
-      payload->mutable_claims();
+// Extract JWT audience into the JwtPayload.
+// Return true if "aud" is extracted. Otherwise, return false.
+bool ExtractJwtAudience(const Envoy::Json::Object& obj,
```

Comment:
Do you need to return bool ?

Reply:
Returning bool provides more information, i.e., whether audience is extracted. Though currently such information is not used, it makes this function more future-proof.

Reply:
There is no point to have the feature if you don't use it yet, it creates confusion for reader. Plus, the function is in the file scope only, so it's not much a hassle to change it when you need it.

Reply:
Done: changed returning bool to void.